src/components/date/date.test.jsx

```diff
@@ -0,0 +1,376 @@
+  });
+
+  it('updates the calendar when the user types', async () => {
+    const date = getLocaleDate(new window.Date('1999-01-01'));
+
+    render(<Date {...requiredProps} />);
+    userEvent.type(screen.getByLabelText('Test label'), '1999-01-01');
+    expect(await screen.findByText(date.calendarHeading)).toBeInTheDocument();
+  });
+
+  describe('calendar behavior', () => {
+    it('toggles open/close', () => {
+      const today = getLocaleDate(new window.Date());
+
+      render(<Date {...requiredProps} />);
+      userEvent.click(screen.getByTitle('Test label calendar button'));
+      expect(screen.getByText(today.calendarHeading)).toBeInTheDocument();
+      userEvent.click(screen.getByTitle('Test label calendar button'));
+      expect(screen.queryByText(today.calendarHeading)).not.toBeInTheDocument();
+    });
+
+    it('closes on selection', () => {
+      const today = getLocaleDate(new window.Date());
+
+      render(<Date {...requiredProps} />);
+      userEvent.click(screen.getByTitle('Test label calendar button'));
+      userEvent.click(screen.getByLabelText(today.calendarDate));
+      expect(screen.queryByText(today.calendarDate)).not.toBeInTheDocument();
+      expect(screen.getByLabelText('Test label')).toHaveValue(today.displayValue);
+    });
+
+    it('closes on blur', () => {
+      const today = getLocaleDate(new window.Date());
+
+      render(<Date {...requiredProps} />);
+      userEvent.click(screen.getByTitle('Test label calendar button'));
+      expect(screen.getByText(today.calendarHeading)).toBeInTheDocument();
+      userEvent.click(document.body);
+      expect(screen.queryByText(today.calendarHeading)).not.toBeInTheDocument();
+      expect(document.activeElement).not.toBe(screen.getByLabelText('Test label'));
+    });
+
+    it('closes on escape', () => {
+      const today = getLocaleDate(new window.Date());
+
+      render(<Date {...requiredProps} />);
+      userEvent.type(screen.getByLabelText('Test label'), '');
+      expect(screen.getByText(today.calendarHeading)).toBeInTheDocument();
+      userEvent.type(screen.getByLabelText('Test label'), '{esc}', { skipClick: true });
+      expect(screen.queryByText(today.calendarHeading)).not.toBeInTheDocument();
+    });
+
+    it('does not open the native calendar picker', () => {
+      const onClickSpy = jest.fn(event => event.persist());
+      const onKeyDownSpy = jest.fn(event => event.persist());
+
+      render((
+        <div onClick={onClickSpy} onKeyDown={onKeyDownSpy}>
+          <Date {...requiredProps} />
+        </div>
+      ));
+
+      userEvent.click(screen.getByLabelText('Test label'));
+      userEvent.type(screen.getByLabelText('Test label'), ' ');
+
+      expect(onClickSpy).toBeCalledWith(expect.objectContaining({
+        defaultPrevented: true,
+        target: expect.anything(),
+      }));
+      expect(onKeyDownSpy).toBeCalledWith(expect.objectContaining({
```

Comment:
Is there another expectation we can write to ensure that some markup doesn't exist in the page?

Reply:
Nah, native calendar pickers are OS-based. There is nothing on the DOM to indicate it is open since the browser communicates with the OS via the event (or a system call if the event's default has not been prevented).